### PR TITLE
python3Packages.aiohttp-socks: 0.8.1 -> 0.8.4

### PR DESCRIPTION
**Pull request by mbalatsko**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mbalatsko** [comment]
Result of `nixpkgs-review pr 259006` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>30 packages built:</summary>
  <ul>
    <li>electrum</li>
    <li>electrum-grs</li>
    <li>electrum-grs.dist</li>
    <li>electrum-ltc</li>
    <li>electrum-ltc.dist</li>
    <li>electrum.dist</li>
    <li>maigret</li>
    <li>maigret.dist</li>
    <li>matrix-commander</li>
    <li>matrix-commander.dist</li>
    <li>mirage-im</li>
    <li>opsdroid</li>
    <li>opsdroid.dist</li>
    <li>pantalaimon</li>
    <li>pantalaimon-headless</li>
    <li>pantalaimon-headless.dist</li>
    <li>pantalaimon.dist</li>
    <li>python310Packages.aiogram</li>
    <li>python310Packages.aiogram.dist</li>
    <li>python310Packages.aiohttp-socks</li>
    <li>python310Packages.aiohttp-socks.dist</li>
    <li>python310Packages.matrix-nio</li>
    <li>python310Packages.matrix-nio.dist</li>
    <li>python311Packages.aiogram</li>
    <li>python311Packages.aiogram.dist</li>
    <li>python311Packages.aiohttp-socks</li>
    <li>python311Packages.aiohttp-socks.dist</li>
    <li>python311Packages.matrix-nio</li>
    <li>python311Packages.matrix-nio.dist</li>
    <li>weechatScripts.weechat-matrix</li>
  </ul>
</details>

File: pkgs/development/python-modules/aiohttp-socks/default.nix [hunk 1]
```diff
@@ -1,26 +1,48 @@
-{ lib, fetchPypi, buildPythonPackage, pythonOlder, aiohttp, python-socks, attrs }:
+{ lib
+, fetchFromGitHub
+, buildPythonPackage
+, pythonOlder
+, aiohttp
+, python-socks
+, attrs
```
Comment (austinbutler) [inline pkgs/development/python-modules/aiohttp-socks/default.nix:7]: If it's not required we can remove it here.

```suggestion
```

File: pkgs/development/python-modules/aiohttp-socks/default.nix [hunk 2]
```diff
@@ -1,26 +1,48 @@
...
-  version = "0.8.1";
+  version = "0.8.4";
 
-  src = fetchPypi {
-    inherit version;
-    pname = "aiohttp_socks";
-    hash = "sha256-duWEJDS5Ts3EWNRZ8MJcD7buMh3FRKA+bJiO3P7QWz0=";
+  disabled = pythonOlder "3.6";
+
+  src = fetchFromGitHub {
```
Comment (austinbutler) [inline pkgs/development/python-modules/aiohttp-socks/default.nix:22]: Why is this change from `fetchPypi` to `fetchFromGithub` necessary?

File: pkgs/development/python-modules/aiohttp-socks/default.nix [hunk 3]
```diff
@@ -1,26 +1,48 @@
...
+    owner = "romis2012";
+    repo = "aiohttp-socks";
+    rev = "v${version}";
+    hash = "sha256-KQeZxwxvWvjS24DoS6G/AIZ9tCpZZtnpJ8adVB+TQpA=";
```
Comment (austinbutler) [inline pkgs/development/python-modules/aiohttp-socks/default.nix:26]: Locally it seems I get a different hash.

**Review by austinbutler — CHANGES_REQUESTED** [review]

**Comment by wegank** [comment]
Already on staging-next: https://github.com/NixOS/nixpkgs/commit/df2813e2d2882d7e2977be602a684a03afd2bc9f

**Pull request closed without merging**